mitiq/__init__.py

```diff
@@ -13,6 +13,9 @@
     QuantumResult,
 )
 
+from mitiq.utils import (
+    tensor_product, matrix_to_vector, _safe_sqrt, vector_to_matrix)
+
```

Comment:
What about if we remove this? The reason is to keep the root of `mitiq` free from very specific and technical functions.

We can still access utility functions with `from mitiq.utils import xxxx` instead of using the shortcut `from mitiq import xxxx` .